pytorch_lightning/trainer/data_loading.py

```diff
@@ -159,62 +157,74 @@ def _resolve_batch_sampler(dl_args, dataloader, sampler, mode: Optional[RunningS
             )
             if is_predicting:
                 batch_sampler = IndexBatchSamplerWrapper(batch_sampler)
-            dl_args['batch_sampler'] = batch_sampler
-            dl_args['batch_size'] = 1
-            dl_args['shuffle'] = False
-            dl_args['sampler'] = None
-            dl_args['drop_last'] = False
-        else:
-            dl_args['sampler'] = sampler
-            dl_args['shuffle'] = False
-            dl_args['batch_sampler'] = None
-
-        return dl_args
+            return {
+                'sampler': None,
+                'shuffle': False,
+                'batch_sampler': batch_sampler,
+                'batch_size': 1,
+                'drop_last': False,
+            }
+        return {
+            'sampler': sampler,
+            'shuffle': False,
+            'batch_sampler': None,
+        }
 
     def replace_sampler(self, dataloader: DataLoader, sampler, mode: Optional[RunningStage] = None) -> DataLoader:
-        skip_keys = ('sampler', 'batch_sampler', 'dataset_kind')
-        skip_signature_keys = ('args', 'kwargs', 'self')
-
+        # get the dataloader instance attributes
         attrs = {k: v for k, v in vars(dataloader).items() if not k.startswith("_")}
-
-        params = set(inspect.signature(dataloader.__init__).parameters)
-        contains_dataset = True
-
-        if type(dataloader) is not DataLoader:
```

Comment:
can we still have a check if this is at least a subclass of the DataLoader? Ideally we can fail early if this is not a subclass of a DataLoader at all (i.e. then we cannot assume anything on the internal structure, especially not that most of the init args correspond to an attribute).

Reply:
Done!